File: frontend/src/service/leaveApprovalService.js
import { DOMAIN } from "../utils/setting/config";
import authorizedAxiosInstance from "./authorizedAxios";

const getAllSubLeaveRequest = async (
  page = 0,
  size = 10,
  startCreatedAt = "",
  endCreatedAt = "",
  leaveDateStart = "",
  leaveDateEnd = "",
  leaveTypeId = 0,
  statusId = 0,
  sort = "idRequest,desc",
  employeeName
) => {
  const params = {
    size,
    page,
    startCreatedAt,
    endCreatedAt,
    leaveDateStart,
    leaveDateEnd,
    leaveTypeId,
    statusId,
    sort,
    employeeName,
  };
  return await authorizedAxiosInstance.get(`${DOMAIN}/leave_approval`, {
    params,
  });
};

const processLeaveRequestAPI = async (idRequest, noteProcess, action) => {
  return await authorizedAxiosInstance.post(
    `${DOMAIN}/leave_approval/process`,
    {
      idRequest,
      noteProcess,
      action,
    }
  );
};

export const leaveApprovalService = {
  getAllSubLeaveRequest,
  processLeaveRequestAPI,
};

File: backend/src/main/java/quyet/leavemanagement/backend/service/LeaveApprovalService.java
package quyet.leavemanagement.backend.service;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;
import quyet.leavemanagement.backend.dto.response.leave_request.LeaveRequestResponse;
import quyet.leavemanagement.backend.dto.response.my_leave_request.MyLeaveRequestResponse;

import java.util.List;

@Service
public interface LeaveApprovalService {

    Page<LeaveRequestResponse> filterAllLeaveRequests(
            String startCreatedAt, String endCreatedAt, String leaveDateStart, String leaveDateEnd,
            int leaveTypeId, int statusId, String employeeName, Pageable pageable
    );

    LeaveRequestResponse processLeaveRequest(Integer idRequest, String noteProcess, String action);
}
